SW_VegProd.c

```diff
@@ -841,3 +865,541 @@ void get_critical_rank(void){
+                     "sum to more than user defined total grass cover.");
+        }
+        // Find indices to estimate related to grass (i.e., C3, C4 and annual grasses)
+        for(index = C3Index; index < grassAnn; index++) {
+            if(inputValues[index] == SW_MISSING) {
+                grassesEstim[grassEstimSize] = index;
+                grassEstimSize++;
+            }
+        }
+
+        // Check if totalSumGrasses is greater than zero
+        if(totalSumGrasses > 0) {
+
+            // Check if there is only one grass index to be estimated
+            if(grassEstimSize == 1) {
+                // Set element to SumGrassesFraction - inputSumGrasses
+                estimCover[grassesEstim[0]] = SumGrassesFraction - inputSumGrasses;
+
+                // Set totalSumGrasses to zero
+                totalSumGrasses = 0.;
+            }
+        } else {
+            // Otherwise, totalSumGrasses is zero or below
+            for(index = 0; index < grassEstimSize; index++) {
+                // Set all found ids to estimate to zero
+                estimCover[grassesEstim[index]] = 0.;
+            }
+        }
+    }
+
+    uniqueIndices(isetIndices, iFixed, 3, iFixedSize, iFixed, &iFixedSize);
+
+    // Check if number of elements to estimate is less than or equal to 1
+    if(overallEstimSize <= 1) {
+        if(overallEstimSize == 0) {
+            // Check if we want to fill gaps in data with bare ground
+            if(fillEmptyWithBareGround) {
+                // Set estimCover at index `bareGround` to 1 - (all values execpt
+                // at index `bareGround`)
+                estimCover[bareGround] = 1 - (initialVegSum - estimCover[bareGround]);
+            } else if(initialVegSum < 1) {
+                LogError(logfp, LOGFATAL, "'estimate_PotNatVeg_composition': "
+                         "User defined relative abundance values are all fixed, "
+                         "but their sum is smaller than 1 = full land cover.");
+            }
+        } else if(overallEstimSize == 1) {
+            estimCover[overallEstim[0]] = 1 - initialVegSum;
+        }
+    } else {
+
+        if(PPT_cm * 10 <= 1) {
+            for(index = 0; index < nTypes - 1; index++) {
+                estimCover[index] = 0.;
+            }
+            estimCover[bareGround] = 1.;
+        } else {
+            // Set months of winter and summer (northern/southern hemisphere)
+            // and get their three month values in precipitation and temperature
+            if(inNorthHem) {
+                for(index = 0; index < 3; index++) {
+                    winterMonths[index] = (index + 11) % MAX_MONTHS;
+                    summerMonths[index] = (index + 5);
+                    summerMAP += PPTMon_cm[summerMonths[index]];
+                    winterMAP += PPTMon_cm[winterMonths[index]];
+                }
+            } else {
+                for(index = 0; index < 3; index++) {

```

Comment:
Why are the lines 1115-1119 not covered by tests? I thought that we had now test cases for the southern hemisphere?